skytemple_files/patch/handler/ai_item_improvements.py

```diff
@@ -0,0 +1,204 @@
+)
+from skytemple_files.patch.category import PatchCategory
+from skytemple_files.patch.handler.abstract import AbstractPatchHandler, DependantPatch
+from skytemple_files.common.i18n_util import f, _
+
+MODIFIED_INSTRUCTION = 0xE8BD83F8  # ldmia all the shit
+OFFSET_EU = 0x231F29C - 0x22DCB80  # location of the instruction minus starting point of overlay 29
+OFFSET_US = 0x231E834 - 0x22DC240  # same but US HeHeHaHa
+
+ADD_TYPES_ADDRESS_NA = 0x2EAA4
+ADD_TYPES_ADDRESS_EU = 0x2EBD8
+ADD_TYPES_ORIGINAL_CODE = 0xE3A00024
+
+
+# the function where all the magic happens
+def set_item_ai_flags(item_p_model, item_id: int, ai_flag_1: bool, ai_flag_2: bool, ai_flag_3: bool):
+    item = item_p_model.item_list[item_id]
+    item.ai_flag_1 = ai_flag_1
+    item.ai_flag_2 = ai_flag_2
+    item.ai_flag_3 = ai_flag_3
+
+
+class AiItemImprovementsHandler(AbstractPatchHandler, DependantPatch):
+    @property
+    def name(self) -> str:
+        return "AiItemImprovements"
+
+    @property
+    def description(self) -> str:
+        return _("Tweaks team and enemy AI to use more types of items, and use them more effectively.")
+
+    @property
+    def author(self) -> str:
+        return "happylappy"
+
+    @property
+    def version(self) -> str:
+        return "1.0.0"
+
+    def depends_on(self) -> list[str]:
+        return []
+
+    @property
+    def category(self) -> PatchCategory:
+        return PatchCategory.NEW_MECHANIC
+
+    def is_applied(self, rom: NintendoDSRom, config: Pmd2Data) -> bool:
+        patch_area = get_binary_from_rom(rom, config.bin_sections.overlay29)
+        if config.game_version == GAME_VERSION_EOS:
+            if config.game_region == GAME_REGION_US:
+                return read_u32(patch_area, OFFSET_US) == MODIFIED_INSTRUCTION
+            if config.game_region == GAME_REGION_EU:
+                return read_u32(patch_area, OFFSET_EU) == MODIFIED_INSTRUCTION
+        raise NotImplementedError()
+
+    def apply(self, apply: Callable[[], None], rom: NintendoDSRom, config: Pmd2Data) -> None:
+        param = self.get_parameters()
+        # do an is_applied check for AddTypes. let's say the result is stored to bool add_types_applied
+        if config.game_version == GAME_VERSION_EOS:
+            if config.game_region == GAME_REGION_US:
+                add_types_applied = (
+                    read_u32(rom.loadArm9Overlays([29])[29].data, ADD_TYPES_ADDRESS_NA) != ADD_TYPES_ORIGINAL_CODE
+                )
+            elif config.game_region == GAME_REGION_EU:
+                add_types_applied = (
+                    read_u32(rom.loadArm9Overlays([29])[29].data, ADD_TYPES_ADDRESS_EU) != ADD_TYPES_ORIGINAL_CODE
+                )
+            else:
+                raise NotImplementedError()
+        else:
+            raise NotImplementedError()
+        if add_types_applied:
+            param["AddTypesApplied"] = 1
+        else:
+            param["AddTypesApplied"] = 0
+        self.__parameters = param
```

Comment:
Since you're normally not supposed to overwrite this, I'd add a comment explaining why your're doing it